### PR TITLE
chore(deps): bump deps

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Bump most dependencies

Reduce the number of dependencies to be built by about 30

build: 795 -> 767
build-release: 986 -> 957
test: 883 -> 852
lint: 1118 -> 1086

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #issue

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
Maybe we still need to update the `Cargo.lock`.

File: common/metrics/src/recorder.rs [hunk 1]
```diff
@@ -87,7 +87,9 @@ fn init_prometheus_recorder() {
     let recorder = PrometheusBuilder::new().build_recorder();
     let mut h = PROMETHEUS_HANDLE.as_ref().write();
     *h = Some(recorder.handle());
-    metrics::clear_recorder();
+    unsafe {
```
Comment (BohuTANG) [inline common/metrics/src/recorder.rs:90]: Why need unsafe here?

Comment (PsiACE) [reply common/metrics/src/recorder.rs:90]: https://github.com/metrics-rs/metrics/pull/302

Comment (PsiACE) [reply common/metrics/src/recorder.rs:90]: > `metrics::clear_recorder();` was removed from metrics (technically just marked unsafe and hidden from docs) due to being dangerous to use.

There is a potential data race here, and I think we can open an issue and try to fix it.

**Comment by ygf11** [comment]
Wow, nice work!

I am curious， how you find these crates which need dump and also the target versions?

**Comment by PsiACE** [comment]
> how you find these crates which need dump and also the target versions?

I have written such a document https://databend.rs/doc/faq/routine-maintenance 

The steps are simple and only require some manual tweaks.

**Comment by ygf11** [comment]
> have written such a document https://databend.rs/doc/faq/routine-maintenance

Thanks, learn much :)

**Pull request merged**